pyarlo/camera.py

```diff
@@ -132,95 +135,112 @@ def get_battery_level(self):
     def get_brightness(self):
         """Get the brightness property of camera."""
-        base = self._session.base_stations[0]
-        props = base.get_camera_properties
-        if not props:
+        properties = self.properties
+        if not self.properties:
             return None
-        for cam in props:
-            if cam['serialNumber'] == self.device_id:
-                return cam['brightness']
 
-        return None
+        return properties.get("brightness")
 
     @property
     def get_mirror_state(self):
         """Get the brightness property of camera."""
-        base = self._session.base_stations[0]
-        props = base.get_camera_properties
-        if not props:
+        properties = self.properties
+        if not self.properties:
             return None
-        for cam in props:
-            if cam['serialNumber'] == self.device_id:
-                return cam['flip']
 
-        return None
+        return properties.get("flip")
 
     @property
     def get_flip_state(self):
         """Get the brightness property of camera."""
-        base = self._session.base_stations[0]
-        props = base.get_camera_properties
-        if not props:
+        properties = self.properties
+        if not self.properties:
             return None
-        for cam in props:
-            if cam['serialNumber'] == self.device_id:
-                return cam['mirror']
 
-        return None
+        return properties.get("mirror")
 
     @property
     def get_powersave_mode(self):
         """Get the brightness property of camera."""
-        base = self._session.base_stations[0]
-        props = base.get_camera_properties
-        if not props:
+        properties = self.properties
+        if not self.properties:
             return None
-        for cam in props:
-            if cam['serialNumber'] == self.device_id:
-                return cam['powerSaveMode']
 
-        return None
+        return properties.get("powerSaveMode")
 
     @property
     def is_camera_connected(self):
         """Connectivity status of Cam with Base Station."""
-        base = self._session.base_stations[0]
-        props = base.get_camera_properties
-        if not props:
+        properties = self.properties
+        if not self.properties:
             return None
-        for cam in props:
-            if cam['serialNumber'] == self.device_id:
-                return bool(cam['connectionState'] == 'available')
 
-        return None
+        return bool(properties.get("connectionState") == "available")
 
     @property
     def get_motion_detection_sensitivity(self):
         """Sensitivity level of Camera motion detection."""
-        base = self._session.base_stations[0]
-        props = base.get_camera_properties
-        if not props:
-            return None
-        for cam in props:
-            if cam['serialNumber'] == self.device_id:
-                this_cam = cam
-        triggers = this_cam['capabilities'][9]['Triggers']
```

Comment:
the `Triggers` object wasn't living in the 9th slot of my `capabilities` array, so i tried to make this a bit smarter... i also extracted some of the logic into `self.capabilities` and `self.triggers` properties for future use.

Reply:
:+1: